test/test-lib/github-mock.ts

```diff
@@ -0,0 +1,52 @@
+import * as Octokit from '@octokit/rest';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as mockery from 'mockery';
+
+import getContent = require('../fixtures/github/getContent.json');
+
+class GHMock {
+	repos = {
+		getContents(
+			params?: Octokit.RequestOptions & Octokit.ReposGetContentsParams,
+		): Promise<Octokit.Response<Octokit.ReposGetContentsResponse>> {
+			if (!params) {
+				return Promise.reject(
+					new Error(`gh mock: No parameters provided for get content call.`),
+				);
+			}
+			const contentKey = `${params.owner}/${params.repo}/${params.path}`;
+
+			if (!_.has(getContent, contentKey)) {
+				return Promise.reject(
+					new Error(
+						`gh mock: getContents could not find a mock for ${contentKey}`,
+					),
+				);
+			}
+
+			const response: Octokit.Response<Octokit.ReposGetContentsResponse> = {
```

Comment:
Maybe we could just type it as 
```suggestion
			const response: Pick<Octokit.Response<Octokit.ReposGetContentsResponse>, 'data' | 'status' | 'headers'> = {
```
and
* add a comment that these are the only props that we need from the mock
* drop the extra `as Octokit.Response<Octokit.ReposGetContentsResponse>`